default.nix

```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  rustPlatform,
+
+  coreutils,
+  pkg-config,
+  openssl,
+  perl,
+}:
+rustPlatform.buildRustPackage {
+  name = "omni";
+  src = lib.cleanSource ./.;
+
+  cargoLock.lockFile = ./Cargo.lock;
+
+  nativeBuildInputs = [
+    pkg-config
+    perl
+  ];
+
+  buildInputs = [
+    openssl
+  ];
+
+  propagatedBuildInputs = [
+    coreutils
+  ];
+
+  checkPhase = ''
+    cargo test -- \
+      --skip internal::cache::up_environments::tests::up_environment::test_new_and_init \
```

Comment:
Nix tests run inside an internet-isolated environment, which is why (I believe?) these tests failed. For now I just skipped them.